File: packages/client_react/src/types/material.ts
export interface MaterialDto {
  id: number;
  name: string;
  code: string;
  description?: string | null;
  normId?: number | null;
  normName?: string | null;
  isProduct: boolean;
  isRawMaterial: boolean;
  isReagent: boolean;
  isObsolete: boolean;
  dateCreated: string;
  dateObsolete?: string | null;
  commentsObsolete?: string | null;
}

export interface CreateMaterialDto {
  name: string;
  code: string;
  description?: string | null;
  normId?: number | null;
  isProduct: boolean;
  isRawMaterial: boolean;
  isReagent: boolean;
  isObsolete: boolean;
}

export interface UpdateMaterialDto extends CreateMaterialDto {}

export interface MaterialTestDto {
  id: number;
  name: string;
  code: string;
  typeId: number;
  isArray: boolean;
  typeName: string;
  nrOrd: number;
}

export interface UpdateMaterialTestsDto {
  testIds: number[];
}

export interface MaterialControlCodeDto {
  id: number;
  code: string;
}

File: server/QCLab/Models/Material.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class Material
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public string? Description { get; set; }

    public long? NormId { get; set; }

    public bool IsProduct { get; set; }

    public bool IsRawMaterial { get; set; }

    public bool IsReagent { get; set; }

    public string? CasNumber { get; set; }

    public DateTime DateCreated { get; set; }

    public bool IsObsolete { get; set; }

    public DateTime? DateObsolete { get; set; }

    public string? CommentsObsolete { get; set; }

    public virtual ICollection<ControlCode> ControlCodes { get; set; } = new List<ControlCode>();

    public virtual Norm? Norm { get; set; }

    public virtual ICollection<Spec> Specs { get; set; } = new List<Spec>();

    public virtual ICollection<Test> Tests { get; set; } = new List<Test>();

    public virtual ICollection<Test> TestsNavigation { get; set; } = new List<Test>();
}
